paper/paper.md

```diff
@@ -34,23 +34,40 @@ preferred-citation: article
+Agent-based modeling is a powerful approach for studying complex systems across many disciplines, from economics and sociology to ecology and epidemiology. As simulations grow more sophisticated, researchers need frameworks that can efficiently handle complex agents and their environments, while remaining approachable and flexible. While established platforms like NetLogo and MASON exist, there is a clear need for a modern, Python-based framework that integrates with the scientific Python ecosystem and provides robust ABM capabilities.
 
 Mesa addresses this need by offering a modular, extensible framework that leverages Python's strengths in scientific computing and data analysis. It provides a comprehensive set of tools for creating, running, and analyzing agent-based models while maintaining Python's emphasis on readability and simplicity.
 
 # Core capabilities
+Agent-based models, or artificial societies, are composed of autonomous heteregouneous agents that are positioned in one or more space(s). Given a space, agents have *local* interactions with their neighbors. The aggregate dynamics of a system under study emerges from these local interactions (Epstein, chapter 2, AGENT-BASED COMPUTATIONAL MODELS AND GENERATIVE SOCIAL SCIENCE; Epstein Axtel (1996)). That is, "*situate an initial population of autonomous heterogeneous agents in a relevant spatial environment; allow them to interact according to simple local rules, and thereby generate—or “grow”—the macroscopic regularity from the bottom up*" (Epstein, axtel 1996; add page number!).
+
+
 Mesa is implemented in pure Python (3.11+) with a modular architecture separating:
-1. Core ABM components (agents, spaces, model management)
-2. Data collection and analysis
+1. Core ABM components (*i.e.,* agents, spaces, agent activation, control over random numbers)
+2. Data collection and support for model experimentation
 3. Visualization systems
 
 This design allows selective use of components while enabling extension and customization. The framework integrates with the scientific Python ecosystem including NumPy, pandas, and Matplotlib.
 
 ## Core ABM components
+
+some prelim stuff on rng and the model class
+
+
+### Agents
+
+* the agent is a central concept in ABM
+* in mesa it is a class that is designed to be subclassed by the user
+* basic structure: `__init__` and `__step__`
+* various subclasses are availble for more sophisticated functionality
+* the risk of memory leaks, so agents automatigcally register themselves with the model and have a remove method
+
 ### Agent management
+since memory leaks are key thing, agent sets internally use weakrefs
+
```

Comment:
While definitely interesting, it's an implementation detail, so I'm not sure it's suited for such a short paper.
```suggestion
```

Reply:
I disagree on this point. In fact, memory management is one of the overriding concerns in ABM frameworks, so we should mention it. Length is not really an argument because this can all be handled in 3 or 4 sentences (and, of course, only once and not twice, as is currently the case).